xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -569,7 +573,7 @@ private void handleLdsResponse(DiscoveryResponse ldsResponse) {
    * invalid data for gRPC's usage. Otherwise, an ACK request is sent to management server.
    */
   private void handleRdsResponse(DiscoveryResponse rdsResponse) {
-    logger.log(Level.FINE, "Received an RDS response: {0}", rdsResponse);
+    logger.log(Level.FINE, "Received an RDS response: {0}", respPrinter.print(rdsResponse));
```

Comment:
Because `respPrinter.print()` is heavy, check `isLoggable` first. Otherwise LGTM.
```java
if (logger.isLoggable(Level.FINE)) {
  logger.log(Level.FINE, "Received an RDS response: {0}", respPrinter.print(rdsResponse));
}
```

Reply:
Sure, change. Btw, I will be working on improving general xDS logging and the logging mechanism will be changed. This PR mainly implements the proto printer, which is needed anyway.